src/ServerConfigChecker.cpp

```diff
@@ -71,32 +83,30 @@ void	ServerConfig::checker() {
 			else if (VERBOSE)
 				std::cout << "no]" << std::endl;
 
-			if (!((*jit).auto_index) && (*jit).index.empty()) throw std::runtime_error("missing index config");
+			if ( !((*jit).auto_index ) && (*jit).index.empty() && !((*jit).is_redirect) ) throw std::runtime_error("missing index config");
 			if ((*jit).auto_index && VERBOSE) std::cout << "\t\tRoute auto_index\n\t\t\t[yes]" << std::endl;
 
 			if ((*jit).auto_index && !((*jit).index.empty())) throw std::runtime_error("both auto_index and manual index set");
 			if (VERBOSE)
 				std::cout << "\t\tRoute index:\n\t\t\t[" << (*jit).index << "]" << std::endl;
 
-			if ((*jit).accepted_methods.empty()) {	
-				(*jit).accepted_methods.push_back("GET");
-				(*jit).accepted_methods.push_back("POST");
-				(*jit).accepted_methods.push_back("PATCH");
-				(*jit).accepted_methods.push_back("PUT");
-			}
-			else {
+			if (!((*jit).accepted_methods.empty())) {	
 				if (VERBOSE)
 					std::cout << "\t\tRoute accepted methods:" << std::endl;
 				for (std::vector<std::string>::iterator vit = (*jit).accepted_methods.begin(); vit != (*jit).accepted_methods.end(); vit++) {
 					if (VERBOSE)
 						std::cout << "\t\t\t[" << *vit << "]" << std::endl;
-					if ((*vit).compare("GET") != 0 && (*vit).compare("POST") != 0 && (*vit).compare("PATCH") != 0 && (*vit).compare("PUT") != 0 )
-						throw std::runtime_error("Error: invalid route method");
 				}
+			} else { // If not accepted methods are set, set all available
+					(*jit).accepted_methods.push_back("GET");
+    				(*jit).accepted_methods.push_back("POST");
+    				(*jit).accepted_methods.push_back("PUT");
+    				(*jit).accepted_methods.push_back("PATCH");
+    				(*jit).accepted_methods.push_back("DELETE");
+					(*jit).accepted_methods.push_back("HEAD");
+					// More to be set
```

Comment:
These are not all methods available. A request method can be anything you want, although usually these are the only ones that are implemented on servers. This is not necessary and should be removed. More info: https://www.rfc-editor.org/rfc/rfc9110.html#name-overview

Reply:
With #19 I meant implementing `methods` directive in the config. (in the subject file: `Define a list of accepted HTTP methods for the route.`) Which should only accept the request if the method is listed in the `methods` directive. In HTTP, methods are not just a predefined list of methods (Although there are some standard methods) and making a request with the request method `ABC` is also valid (Although probably not implemented). Therefore we can not possibly add all possible methods to the list (as there is literally an infinite amount of methods). So we should check if the list is empty, and if not we should check if the method is in the list.